app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,85 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:ordered, :index]
+
+  def index
+    @merchant_orders = []
+    @merchant_products = []
+
+    Order.all.each do |order|
+      order.order_items.each do |item|
+        merchant_product = Product.find_by(id: item.product_id)
+
+        if session[:user_id] == merchant_product.user_id
```

Comment:
This logic seems like it could go into the merchant model so that you could use something like `merchant.orders`